components/contribution-flow/PaymentMethodList.tsx

```diff
@@ -121,14 +129,20 @@ export default function PaymentMethodList(props: PaymentMethodListProps) {
   });
 
   const hostSupportedPaymentMethods = props.host?.supportedPaymentMethods ?? [];
-  const [paymentIntent, stripe, loadingPaymentIntent] = usePaymentIntent({
+  const [paymentIntent, stripe, loadingPaymentIntent, paymentIntentCreateError] = usePaymentIntent({
     skip: !hostSupportedPaymentMethods.includes(PaymentMethodLegacyType.PAYMENT_INTENT),
     amount: { valueInCents: props.stepDetails.amount, currency: props.stepDetails.currency },
     fromAccount: props.fromAccount.isGuest
       ? undefined
       : typeof props.fromAccount.id === 'string'
       ? { id: props.fromAccount.id }
       : { legacyId: props.fromAccount.id },
+    guestInfo: {
+      email: props.stepProfile?.email,
+      name: props.stepProfile?.name,
+      location: props.stepProfile?.location,
+      captcha: props.stepProfile?.captcha,
+    },
```

Comment:
`legalName` is missing there.

Maybe have a common function with https://github.com/opencollective/opencollective-frontend/blob/ffd5f6726d093312e688157d93bc37f073ac6566/components/contribution-flow/index.js#L250 added in `components/contribution-flow/utils.js`. Otherwise it's very likely that, if adding a new field there, developers will forget to add it in both places.

```suggestion
    guestInfo: getGuestInfoFromStepProfile(props.stepProfile),
```